subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/incremental/IncrementalResultStoringCompiler.java

```diff
@@ -89,27 +86,13 @@ private AnnotationProcessingData getAnnotationProcessingResult(JavaCompileSpec s
     }
 
     private AnnotationProcessingData convertProcessingResult(AnnotationProcessingResult processingResult) {
-        Map<String, Set<String>> generatedTypesByOrigin = processingResult.getGeneratedTypesWithIsolatedOrigin();
-        Map<String, Set<GeneratedResource>> generatedResourcesByOrigin = processingResult.getGeneratedResourcesWithIsolatedOrigin();
-        Set<String> aggregatedTypes = processingResult.getAggregatedTypes();
-        Set<String> aggregatingTypes = processingResult.getGeneratedAggregatingTypes();
-        Set<GeneratedResource> aggregatingResources = processingResult.getGeneratedAggregatingResources();
-        return new AnnotationProcessingData(intern(generatedTypesByOrigin), intern(aggregatedTypes), intern(aggregatingTypes), generatedResourcesByOrigin, aggregatingResources, processingResult.getFullRebuildCause());
-    }
-
-    private Set<String> intern(Set<String> types) {
```

Comment:
We no longer keep this in an in-memory cache, but on disk, so interning is not necessary.